src/librustc_middle/mir/mod.rs

```diff
@@ -80,7 +80,8 @@ pub enum MirPhase {
     Const = 1,
     Validated = 2,
     DropElab = 3,
-    Optimized = 4,
+    OptimizedWithGenerators = 4,
+    Optimized = 5,
```

Comment:
Now that these affect validation, somewhere, there should be an extensive documentation describing how these phases differ in terms of the invariants they place on the MIR.

Reply:
I added some docs